crates/context_aware_config/src/api/dimension/types.rs

```diff
@@ -1,18 +1,72 @@
 use chrono::{DateTime, NaiveDateTime, Utc};
+use derive_more::{AsRef, Deref, DerefMut, Into};
+use regex::Regex;
 use serde::{Deserialize, Deserializer, Serialize};
 use serde_json::Value;
 
 use crate::db::models::Dimension;
 
 #[derive(Debug, Deserialize)]
 pub struct CreateReq {
-    pub dimension: String,
-    pub priority: i32,
+    pub dimension: DimensionName,
+    pub priority: Priority,
     pub schema: Value,
     #[serde(default, deserialize_with = "deserialize_option")]
     pub function_name: Option<Value>,
 }
 
+#[derive(Debug, Deserialize, AsRef, Deref, DerefMut, Into)]
+#[serde(try_from = "i32")]
+pub struct Priority(i32);
+impl Priority {
+    fn validate_data(priority_val: i32) -> Result<Self, String> {
+        if priority_val <= 0 {
+            return Err("Priority should be greater than 0".to_string());
+        } else {
+            Ok(Self(priority_val))
+        }
+    }
+}
+
+impl TryFrom<i32> for Priority {
+    type Error = String;
+    fn try_from(value: i32) -> Result<Self, Self::Error> {
+        Ok(Priority::validate_data(value)?)
+    }
+}
+
+#[derive(Debug, Deserialize, AsRef, Deref, DerefMut, Into)]
+#[serde(try_from = "String")]
+pub struct DimensionName(String);
+impl DimensionName {
+    pub fn validate_data(name: String) -> Result<Self, String> {
+        let regex_str: &str = "^[a-zA-Z0-9-_]([a-zA-Z0-9-_.]{0,254}[a-zA-Z0-9-_])?$";
```

Comment:
lets reuse the previous string itself, once its made as a const

Reply:
Lets keep different regex for different entity so that the regex logic doesn't overlap